tasks/output.yml

```diff
@@ -0,0 +1,30 @@
+- name: Get system subscription status
+  command: subscription-manager status
+  register: oasis_roles_rhsm_subscription
+  changed_when: false
+  failed_when: false
+
+- name: Get system subscribed pools
+  # This always exits 0 whether the system is subscribed or not;
+  # stdout is empty on an unsubbed system.
+  command: subscription-manager list --consumed --pool-only
+  register: oasis_roles_rhsm_pools
+  changed_when: false
+  failed_when: false
+
+- name: Set system repositories default value
+  # We only get a return fact from the rhsm_repository module if the
+  # rhsm_repositories var is set. If that return fact is not defined,
+  # set that repositories value to false to indicate rhsm_repository
+  # was not used on this host.
+  set_fact:
+    oasis_roles_rhsm_repositories:
+      repositories: false
+  when: not oasis_roles_rhsm_repositories is defined
+
+- name: Register Role Output
+  set_fact:
+    oasis_roles_rhsm:
+      subscribed: "{{ oasis_roles_rhsm_subscription.rc == 0 }}"
+      subscribed_pool_ids: "{{ oasis_roles_rhsm_pools.stdout.split() }}"
+      repositories: "{{ oasis_roles_rhsm_repositories.repositories }}"
```

Comment:
If the previous task's `when` clause evaluates to false, won't this result in an error accessing `.repositories` of an undefined? Why not just set the value here to `oasis_roles_rhsm_repositories is defined`?

Reply:
`repositories` will be "The complete list of subscribable repositories known to the system, as well as their enabled status. This value will be `false` if the `rhsm_repositories` input var is not used."

The repository management tasks are already setting the `oasis_roles_rhsm_repositories` fact, but only do so if the `rhsm_respositories` module actually runs. There's no way to run that module as a no-op that I can see (there is no `rhsm_repositories_facts` module, and similarly no `redhat_subscription_facts` module), so I had to get a little bit creative when it came to the output there. It will either be a huge array of every repository's status as given by `rhsm_repositories`, or `false`.

Reply:
Ah, I see better now what's going on. If this value is supposed to be a list, then maybe setting the default up on line 21-22 should be an empty array `[]` instead of `false` when the `rhsm_repositories` module didn't get run?

Reply:
Would that information be important enough to a consumer vs just wanting to know which repos were added? Off the top of my head I can't think of a place I'd want to distinguish between false/empty array for the list of repositories but I definitely have a very limited imagination in this realm since my knowledge of RHSM in general is so limited.

Reply:
I had a bit of time to think about it, and I think the answer is "nope". If it's desired, we can add it on request. I've gone ahead and removed this output for now.